ykcapi/yk.h

```diff
@@ -51,40 +51,37 @@ void yk_location_drop(YkLocation);
 void yk_stopgap(void *addr, uintptr_t size, uintptr_t retaddr, void *rsp);
 
 #if defined(__x86_64)
-void __llvm_deoptimize(void* addr, uintptr_t size) {
+__attribute__((naked)) void __llvm_deoptimize(void* addr, uintptr_t size) {
     // Push all registers to the stack before they can be clobbered, so that we
     // can find their values after parsing in the stackmap. After that we store
     // the value of RSP in a variable so we know where to look for the register
     // values. The order in which we push the registers is equivalent to the
-    // Sys-V x86_64 ABI, which the stackmap format uses as well. However, since
-    // we are interested in the value of RSP in the previous stack frame, which
-    // has changed when we entered the __llvm_deoptimize function, we push RBP
-    // instead, which is equivalent. Note, that because of this there is no
-    // (easy) way to access RBP from the previous frame, so we just assume it's
-    // never referenced by the stackmap.
-    void *rsp;
-    __asm__ volatile (
-            "push %%rbp\n" // old RSP
-            "push %%rbp\n" // placeholder
-            "push %%rdi\n"
-            "push %%rsi\n"
-            "push %%rbx\n"
-            "push %%rcx\n"
-            "push %%rdx\n"
-            "push %%rax\n"
-            "movq %%rsp, %0\n": "=rm" (rsp));
-
-    // We can use this function's return address to lookup the correct record
-    // in the stackmap.
-    void *ret = __builtin_return_address(0);
+    // Sys-V x86_64 ABI, which the stackmap format uses as well. This function
```

Comment:
> This function...

I think this bit can be made shorter.

All we really need to say is that we use a naked function so that the C compiler cannot emit a function prolog that would fiddle with the stack pointer.

You don't need to say anything about asm, as naked functions in C are by definition, purely asm.

Reply:
I hadn't heard of naked functions before and had to look them up. Fortunately I chose my search terms carefully!

Reply:
Try 268e8ce

Reply:
nitpick: "stackframe" or "stack frame"? Sorry to be a pain.